packages/tools/devtools/devtools-view/src/TelemetryUtils.ts

```diff
@@ -48,3 +48,46 @@ export class ConsoleVerboseLogger implements ITelemetryBaseLogger {
 		this.baseLogger?.send(event);
 	}
 }
+
+/**
+ * Key for the local storage entry that stores the usage telemetry opt-in setting.
+ */
+const telemetryOptInKey: string = "fluid:devtools:telemetry:optIn";
+
+/**
+ * Hook for getting and setting the usage telemetry opt-in setting, backed by brower's local storage.
+ * @returns A tuple (React state) with the current value and a setter for the value.
+ */
+export const useTelemetryOptIn = (): [boolean, React.Dispatch<React.SetStateAction<boolean>>] => {
+	const [value, setValue] = React.useState(() => {
+		return getStorageValue(telemetryOptInKey, true);
+	});
+
+	React.useEffect(() => {
+		localStorage.setItem(telemetryOptInKey, value.toString());
+	}, [value]);
+
+	return [value, setValue];
+};
+
+/**
+ * Logger that forwards events to another logger only when the setting to opt-in to usage telemetry is enabled.
```

Comment:
To clarify, events are still generated, but just not sent to the 2nd logger if the user isn't opted in right?

Reply:
Correct. And the way I set it up, they first get to the ConsoleLogger from the other PR, so they'll get written to console (only visible if the console's filter is set to Verbose) either way, and then when they hit this logger they'll get forwarded to the 2nd logger (the "external one") if opt-in is enabled, and dropped otherwise.